### PR TITLE
PFM 1106 - Search Attendance Register API

### PR DESCRIPTION
**Pull request by anitha-srinivasan-egov**
PFM 1106 - Search Attendance Register API - Search Attendance Register API - Query builder, Repository, Validator, Service and Controller.

The reason for two queryBuilder classes for Attendee and Staff is to due to join cardinality while joining three tables.

### CODE REVIEW

File: backend/attendance-service/src/main/java/org/egov/repository/rowmapper/StaffQueryBuilder.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+import org.springframework.stereotype.Component;
+
+import java.util.Collection;
+import java.util.List;
+
+@Component
+public class StaffQueryBuilder {
+
+    private static final String FETCH_ATTENDANCE_REGISTER_QUERY = "SELECT register.*," +
+            "staff.id AS staffId,staff.individual_id AS staffIndId,staff.register_id AS staffRegId,staff.tenantid AS staffTenant,staff.enrollment_date AS staffEnroll,staff.deenrollment_date AS staffDeenroll "+
```
Comment (rohit-eGov) [inline backend/attendance-service/src/main/java/org/egov/repository/rowmapper/StaffQueryBuilder.java:14]: Do we require register.* in this search query as we are using this query to show only staff data

File: backend/attendance-service/src/main/java/org/egov/repository/rowmapper/AttendeeQueryBuilder.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+import org.springframework.stereotype.Component;
+
+import java.util.Collection;
+import java.util.List;
+
+@Component
+public class AttendeeQueryBuilder {
+
+    private static final String FETCH_ATTENDANCE_REGISTER_QUERY = "SELECT register.*," +
+            "attendee.id AS attendeeId,attendee.register_id AS attendeeRegId,attendee.individual_id AS attendeeIndividualId, attendee.enrollment_date AS attendeeEnroll, attendee.tenantid AS attendeeTenant " +
```
Comment (rohit-eGov) [inline backend/attendance-service/src/main/java/org/egov/repository/rowmapper/AttendeeQueryBuilder.java:14]: Do we require register.* in this search query as we are using this query to show only attendee data

**Review by rohit-eGov — CHANGES_REQUESTED** [review]

File: backend/attendance-service/src/main/java/org/egov/repository/rowmapper/StaffQueryBuilder.java (hunk 1 above)
Comment (anitha-srinivasan-egov) [reply backend/attendance-service/src/main/java/org/egov/repository/rowmapper/StaffQueryBuilder.java:14]: We also need to send the register details like id, number

File: backend/attendance-service/src/main/java/org/egov/repository/rowmapper/AttendeeQueryBuilder.java (hunk 2 above)
Comment (anitha-srinivasan-egov) [reply backend/attendance-service/src/main/java/org/egov/repository/rowmapper/AttendeeQueryBuilder.java:14]: We also need to send the register details like id, number, namer etc from the register table. Hence added register.*    
"attendanceRegister": [
    {
      "id": "64e33343-7b4c-4353-9abf-4de8f5bcd732",
      "tenantId": "pb.amritsar",
      "registerNumber": "REG/2022-23/001",
      "name": "Class-10-E Physics",
      "startDate": 1665497225000,
      "endDate": 1665497271000,
      "status": "ACTIVE",

**Review by rohit-eGov — APPROVED** [review]

**Review by rushang7-eGov — APPROVED** [review]

**Pull request merged**